File: appMain/src/main/java/com/cartlc/tracker/data/DataCollectionEquipment.java
/**
 * Copyright 2018, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.data;

import java.util.ArrayList;
import java.util.List;

import timber.log.Timber;

/**
 * Created by dug on 5/10/17.
 */

public abstract class DataCollectionEquipment {
    public long id; // projectId or collectionId
    public List<Long> equipmentListIds = new ArrayList();

    public DataCollectionEquipment(long id) {
        this.id = id;
    }

    public void add(long equipmentId) {
        equipmentListIds.add(equipmentId);
    }

    public List<DataEquipment> getEquipment() {
        ArrayList<DataEquipment> list = new ArrayList();
        for (long e : equipmentListIds) {
            list.add(TableEquipment.getInstance().query(e));
        }
        return list;
    }

    public List<String> getEquipmentNames() {
        List<DataEquipment> equipments = getEquipment();
        if (equipments != null) {
            ArrayList<String> list = new ArrayList();
            for (DataEquipment e : equipments) {
                list.add(e.name);
            }
            return list;
        }
        return null;
    }

    public String toString() {
        StringBuilder sbuf = new StringBuilder();
        boolean first = true;
        for (String name : getEquipmentNames()) {
            if (first) {
                first = false;
            } else {
                sbuf.append(",");
            }
            sbuf.append(name);
        }
        return sbuf.toString();
    }

}

File: appMain/src/main/java/com/cartlc/tracker/model/data/DataCollectionEquipment.kt
/**
 * Copyright 2018, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.model.data

import com.cartlc.tracker.model.table.DatabaseTable

/**
 * Created by dug on 5/10/17.
 */
abstract class DataCollectionEquipment(
        private val db: DatabaseTable,
        var id: Long // projectId or collectionId
) {
    var equipmentListIds: MutableList<Long> = mutableListOf()

    val equipment: List<DataEquipment>
        get() {
            val list = mutableListOf<DataEquipment>()
            for (e in equipmentListIds) {
                val item = db.equipment.query(e)
                if (item != null) {
                    list.add(item)
                }
            }
            return list
        }

    val equipmentNames: List<String>?
        get() {
            val list = mutableListOf<String>()
            for (e in equipment) {
                list.add(e.name)
            }
            return list
        }

    fun add(equipmentId: Long) {
        equipmentListIds.add(equipmentId)
    }

    override fun toString(): String {
        val sbuf = StringBuilder()
        var first = true
        for (name in equipmentNames!!) {
            if (first) {
                first = false
            } else {
                sbuf.append(",")
            }
            sbuf.append(name)
        }
        return sbuf.toString()
    }

}
